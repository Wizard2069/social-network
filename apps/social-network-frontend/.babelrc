{
  "presets": [
    [
      "@nrwl/react/babel",
      {
        "runtime": "automatic"
      }
    ]
  ],
  "plugins": ["istanbul"]
}
