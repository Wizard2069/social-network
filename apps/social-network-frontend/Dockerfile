FROM nginx:1.21.3

WORKDIR /app/

COPY ./apps/social-network-frontend/nginx.conf /etc/nginx/sites-available/default

COPY ./dist/apps/social-network-frontend/ /app/build/
RUN cp -R build/* /usr/share/nginx/html

RUN ln -sf /dev/stdout /var/log/nginx/access.log \
	&& ln -sf /dev/stderr /var/log/nginx/error.log

EXPOSE 80

CMD ["/bin/sh",  "-c",  "envsubst < /usr/share/nginx/html/assets/env.template.js > /usr/share/nginx/html/assets/env.js && exec nginx -g 'daemon off;'"]
