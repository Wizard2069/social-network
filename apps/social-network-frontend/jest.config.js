module.exports = {
  displayName: 'social-network-frontend',
  preset: '../../jest.preset.js',
  transform: {
    '^(?!.*\\.(js|jsx|ts|tsx|css|json)$)': '@nrwl/react/plugins/jest',
    '^.+\\.[tj]sx?$': 'babel-jest'
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/apps/social-network-frontend',
  collectCoverageFrom: [
    'src/app/**/*.ts',
    'src/app/**/*.tsx',
    '!src/**/*.svg',
    '!src/app/store.ts'
  ]
};
