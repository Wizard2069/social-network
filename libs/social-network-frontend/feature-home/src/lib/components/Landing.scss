.auth-landing {
  display: flex;
  min-height: 100vh;

  background: url('https://res.cloudinary.com/htcompany-cloud/image/upload/w_1920/v1635427330/social-network/background/htc-landing_agtlks.png') no-repeat center center;
  background-size: cover;

  .logo {
    flex-basis: 100%;
    padding: 30vh 0 0 15vw;

    h3 {
      font-weight: 600;
      font-size: clamp(1.5rem, 5vw, 6rem);
      color: #5d88de;

      filter: drop-shadow(-4px 4px 0px rgba(75, 101, 226, 0.2));
    }

    p {
      font-weight: 300;
      font-size: 18px;
    }

    @media (max-width: 1280px) {
      padding: 35vh 3vw 0 5vw;
    }
  }

  .auth-wrapper {
    padding: 15vh 15vw 0 0;

    @media (max-height: 800px) {
      padding-top: 3rem;
    }
  }

  @media (max-width: 768px) {
    flex-direction: column;

    .logo {
      display: none;
    }

    .auth-wrapper {
      padding: 2rem 2rem 0;
    }
  }
  @media (max-width: 425px) {
    .auth-wrapper {
      padding: 2rem 1rem 0;
    }
  }
}
