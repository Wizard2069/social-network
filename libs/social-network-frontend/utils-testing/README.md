# social-network-frontend-utils-testing

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test social-network-frontend-utils-testing` to execute the unit tests via [Jest](https://jestjs.io).
