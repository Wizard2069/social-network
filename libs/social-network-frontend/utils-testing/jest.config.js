module.exports = {
  displayName: 'social-network-frontend-utils-testing',
  preset: '../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory:
    '../../../coverage/libs/social-network-frontend/utils-testing',
};
