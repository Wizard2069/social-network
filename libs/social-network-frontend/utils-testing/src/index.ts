export * from './lib/mocks/server';
export * from './lib/testUtils';
export * from './lib/mocks/mockUser';
export * from './lib/mocks/oktaAuth';
