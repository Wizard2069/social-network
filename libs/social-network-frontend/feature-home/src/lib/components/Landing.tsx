import React from 'react';
import { useOktaAuth } from '@okta/okta-react';
import { OktaSignInWidget } from '@sn-htc/social-network-frontend/feature-auth';
import { oktaSignInConfig } from '@sn-htc/social-network-frontend/config-constants';
import './Landing.scss';

const Landing = () => {
  const { oktaAuth } = useOktaAuth();

  const onSuccess = async (tokens) => {
    await oktaAuth.handleLoginRedirect(tokens);
  };

  const onError = (err) => {
    console.error('error logging in', err);
  };

  return (
    <div className='auth-landing' data-test='socivio-landing'>
      <div className='logo'>
        <h3>Socivio</h3>
        <p>Connect and share your moments with people</p>
      </div>
      <OktaSignInWidget
        config={oktaSignInConfig}
        onSuccess={onSuccess}
        onError={onError} />
    </div>
  );
};

export default Landing;
