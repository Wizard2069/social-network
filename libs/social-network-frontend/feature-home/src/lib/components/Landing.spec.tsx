import '@testing-library/jest-dom';
import { screen, wrapper } from '@sn-htc/social-network-frontend/utils-testing';
import Landing from './Landing';

describe('Landing Page', () => {
  it('should render correctly', async () => {
    wrapper(<Landing />, { isAuthenticated: false });

    expect(await screen.findByTestId('socivio-landing')).toBeVisible();
  });
});
