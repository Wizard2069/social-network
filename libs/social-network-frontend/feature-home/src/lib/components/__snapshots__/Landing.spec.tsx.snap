// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page should render correctly 1`] = `
<div
  class="auth-landing"
  data-test="socivio-landing"
>
  <div
    class="logo"
  >
    <h3>
      Socivio
    </h3>
    <p>
      Connect and share your moments with people
    </p>
  </div>
  <div
    class="auth-wrapper"
  >
    <main
      class="auth-container main-container no-beacon"
      data-se="auth-container"
      id="okta-sign-in"
      style="display: none;"
      tabindex="-1"
    />
  </div>
</div>
`;
